### ISSUE TITLE
Compilation issue, stuck at lapack

### ISSUE THREAD

**Issue by MarcMignolet** [mesonbuild/meson#12693]
I have been trying to install tblite on windows but I am stuck with an error regarding lapack. I am attaching the log file. Could someone help?
I am new at this and stuck....
Thank you!
[meson-log.txt](https://github.com/mesonbuild/meson/files/13824059/meson-log.txt)

**Comment by bruchar1** [comment]
The `lapack` dependency was not found. Is it installed on your system?

Are `CMAKE_MODULE_PATH` and / or `PKG_CONFIG_PATH` properly set?

**Comment by MarcMignolet** [comment]
> The `lapack` dependency was not found. Is it installed on your system?

 I have downloaded Windows x64 OpenBLAS binaries via https://sourceforge.net/projects/openblas/files/. Is this sufficient?

> Are `CMAKE_MODULE_PATH` and / or `PKG_CONFIG_PATH` properly set?

This is the procedure for a Unix System, correct? I am using a Windows PC.

**Comment by bruchar1** [comment]
No, this is not sufficient. meson needs a way to find the path to the lib and header files. It can do it either using cmake or pkgconfig. You need to tell cmake or pkgconfig where to find the definitions of the available libraries.

For pkgconfig, it needs a file with .pc extension, and to allow pkgconfig finding it, you must have the path to the containing folder in your PKG_CONFIG_PATH environment variable. You also need the pkgconfig or pkgconf executable somewhere in your path.

For cmake, it is similar, but using .cmake files and CMAKE_MODULE_PATH variable.

For openblas, those files are located respectively in the lib/pkgconfig and lib/cmake folders of the distribution.

**Comment by MarcMignolet** [comment]
> No, this is not sufficient. meson needs a way to find the path to the lib and header files. It can do it either using cmake or pkgconfig. You need to tell cmake or pkgconfig where to find the definitions of the available libraries.
> 
> For pkgconfig, it needs a file with .pc extension, and to allow pkgconfig finding it, you must have the path to the containing folder in your PKG_CONFIG_PATH environment variable. You also need the pkgconfig or pkgconf executable somewhere in your path.
> 
> For cmake, it is similar, but using .cmake files and CMAKE_MODULE_PATH variable.
> 
> For openblas, those files are located respectively in the lib/pkgconfig and lib/cmake folders of the distribution.

Thank you. I have updated the environmental variables. I do not believe I have a pkgconfig or pkgconf executable on my PC. Is it something separate I need to install? It was not listed in the install instructions by either meson (https://mesonbuild.com/SimpleStart.html) or the specific project I want to build (https://tblite.readthedocs.io/en/latest/installation.html).

**Comment by eli-schwartz** [comment]
> Is it something separate I need to install? It was not listed in the install instructions by either meson (https://mesonbuild.com/SimpleStart.html) or the specific project I want to build (https://tblite.readthedocs.io/en/latest/installation.html).

pkg-config isn't a requirement for meson, because not all projects have dependencies -- and therefore don't need pkg-config to do dependency lookup. This is why when a project does specify a dependency, and meson cannot locate the dependency, it emits a log warning that the pkg-config program could not be found.

Some ecosystems for Windows provide pkg-config in an integrated manner. msys2, cygwin, conan, conda, spack...

It's also possible to manually download and install it, pkgconf does a good job of building standalone (it uses meson).

Your log file does show that it also tries to link to -llapack which should work as long as the lapack library is on your compiler search path (for example, in $LDFLAGS).
